ingestion/src/metadata/ingestion/models/patch_request.py

```diff
@@ -192,20 +203,61 @@ def build_patch(
     # for a user editable fields like descriptions, tags we only want to support "add" operation in patch
     # we will remove the other operations for replace, remove from here
     if restrict_update_fields:
-        patch.patch = [
-            patch_ops
-            for patch_ops in patch.patch
+        patch_ops_list = []
+        for patch_ops in patch.patch:
             if _determine_restricted_operation(
-                patch_ops=patch_ops,
-                restrict_update_fields=restrict_update_fields,
-            )
-        ]
-
+                patch_ops=patch_ops, restrict_update_fields=restrict_update_fields
+            ):
+                if (
+                    patch_ops.get("op") == PatchOperation.REPLACE.value
+                    and patch_ops.get("value") is None
+                ):
+                    patch_ops["op"] = PatchOperation.REMOVE.value
+                    del patch_ops["value"]
+                patch_ops_list.append(patch_ops)
+        patch.patch = patch_ops_list
     return patch
 
 
+def _sort_array_entity_fields(
+    source: T,
+    destination: T,
+    array_entity_fields: Optional[List] = None,
+):
+    """
+    Sort the array entity fields to make sure the order is consistent
+    """
+    for field in array_entity_fields:
```

Comment:
@OnkarVO7 how are we sorting it, basically what element are we  choosing to sort on?

Reply:
We are sorting in the same order how the columns are stored on OpenMetadata side

Reply:
@OnkarVO7 any reason to apply this sorting , is the response not coming with that sort order?

Reply:
@harshach yes the response for columns from the source for a few connectors is coming in random order as described in https://github.com/open-metadata/OpenMetadata/issues/15563. Due to this the patch request for the columns list does not get formed correctly

Hence we are sorting the response from source according to the columns order stored on OM side